### DISCUSSION TITLE
Databricks OAuth machine-to-machine (M2M) authentication

### DISCUSSION THREAD

**Discussion by TobiasRasbold** [mullerpeter/databricks-grafana#79] [Ideas]
Hi mullerpeter,

first of all, thanks for your work at this repository!

I would think it would be great feature if this repository also proivdes OAuth using Databricks M2M authentication next to access tokens.
An advantage of the M2M auth against Personal Access Token (PAT) is that is does not expire. 
So refreshing tokens are not needed any more by using M2M auth.
For OAuth M2M only a Databricks Service Principal and it's client_id and secret is required.

Please find more details: https://docs.databricks.com/en/dev-tools/auth/oauth-m2m.html#oauth-machine-to-machine-m2m-authentication.

What's your opinion on that?

Best regards,
Tobias

**Comment by mullerpeter** [comment]
Hi @TobiasRasbold, thanks for raising the issue. Yes adding support for M2M auth is a great idea and I put it on my todo list. Will try to add it soon.

**Comment by mullerpeter** [comment]
Hi @TobiasRasbold 

I added support for M2M OAuth in v1.2.6, let me know of that works for you

**Reply by TobiasRasbold** [reply]
Sorry for the very late response. 
Thanks for the fast implementation. Works like a charm!

Best regards,
Tobias
